src/types.ts

```diff
@@ -40,16 +40,9 @@ export interface Game {
   rounds: Round[];
 }
 
-export interface KnowledgePresence {
-  clubs: boolean;
-  diamonds: boolean;
-  hearts: boolean;
-  spades: boolean;
-}
-
-export interface KnowledgeHighest {
-  clubs: CardRank;
-  diamonds: CardRank;
-  hearts: CardRank;
-  spades: CardRank;
+export interface InfoSuitHighest {
+  clubs?: CardRank;
+  diamonds?: CardRank;
+  hearts?: CardRank;
+  spades?: CardRank;
```

Comment:
That's the main improvement in this PR: the old `KnowledgePresence` can be represented as a subset of `KnowledgeHighest` by just setting the suit to `undefined`